packages/core/src/browser/xhrProxy.ts

```diff
@@ -77,66 +78,72 @@ function proxyXhr() {
   // eslint-disable-next-line @typescript-eslint/unbound-method
   originalXhrAbort = XMLHttpRequest.prototype.abort
 
+  function reportXhr(this: BrowserXHR, clearListeners: () => void) {
```

Comment:
Well, mostly for two reasons:
- Readability: first, that would lead to 3 levels of nested function declarations which I find harder to read
- Performance: like this, we don't recreate a new function each time the function sends is called

Reply:
> Readability

If we go this way, you might want to move it outside of `proxyXhr` function, since it doesn't use anything from it. Also, using `this` to pass the XHR instance is kind of unintuitive, and since `clearListeners` is called unconditionally at the start of the function, I don't see why it should be passed to this function. What about using a normal argument for `xhr` and remove the `clearListeners` argument?

> Performance

This should have a negligeable performance impact.